Content/StructureResolver/StructureResolverExtensionInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\HeadlessBundle\Content\StructureResolver;
+
+use Sulu\Component\Content\Compat\StructureInterface;
+
+interface StructureResolverExtensionInterface
+{
+    public function getKey(): string;
+
+    public function resolve(
+        StructureInterface $requestedStructure,
+        string $locale
+    ): mixed;
```

Comment:
We would more go here with `array` as return value and use beneath `array_merge` so all extension can manipulate different data. This is the case for `breadcrumb` which can be set by multiple extensions e.g. PageBreadcrumb service will do:

```php
if ($structure is not page structure) {
    return [];
}

return [
    'breadcrumbs' => $this->navigationMapper->getBreadcrumb($structure->getUuid()),
];
```

and the ARticleBreadcrumbProvider (living in the article bundle):

```php
if ($structure is not article structure) {
    return [];
}

$parent = $structure->getPageTreeParent();

if (!$parent) {
    return [];
}

return [
    'breadcrumbs' => $this->navigationMapper->getBreadcrumb($parentarent()->getUuid()),
];
```

```suggestion
    /**
     * @return array<string, mixed>
     */
    public function resolve(StructureInterface $requestedStructure, string $locale): array;
```